### PR TITLE
Enqueueing Process creates multiple workpackages per headeraggregation key

### PR DESCRIPTION
**Pull request by metas-ts**
note that 
- `hard_endcoded_hotfix` was just branched from the tag `had_encoded_uat_11`
- i intended to push the changes to a branch `hard_encoded_hotfix_IC_enqueuer` but failed to do so...instead i pushed them to `hard_endcoded_hotfix`

### CODE REVIEW

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/dao/selection/QuerySelectionHelper.java [hunk 1]
```diff
@@ -211,15 +211,14 @@ public <T, ET extends T> TypedSqlQuery<ET> createUUIDSelectionQuery(
...
 				clazz,
 				selectionWhereClause,
 				ctx.getTrxName())
 						.setParameters(querySelectionUUID)
 						.setSqlFrom(selectionSqlFrom)
-						.setOrderBy(selectionOrderBy);
-
-		return querySelection;
+						.setOrderBy(selectionOrderBy)
+				.setOrderBy(SELECTION_LINE_ALIAS);
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/dao/selection/QuerySelectionHelper.java:222]: @metas-ts  why calling setOrderBy twice?

File: backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandidateEnqueuer.java [hunk 2]
```diff
@@ -104,12 +106,14 @@ private IInvoiceCandidateEnqueueResult enqueueSelectionInTrx(
 	{
 		trxManager.assertThreadInheritedTrxExists();
 
-		final Iterable<I_C_Invoice_Candidate> invoiceCandidates = retrieveSelection(pinstanceId);
+		// Here we just need the "set" if ICs and prepare them one by one.
```
Comment (metas-ts) [inline backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandidateEnqueuer.java:109]:
```suggestion
		// Here we just need a"set" of ICs and prepare them one by one...the ordering doesn't matter.
```

**Review by metas-ts — COMMENTED** [review]
lgtm..going to integrate when build is green

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/dao/selection/QuerySelectionHelper.java (hunk 1 above)
Comment (metas-ts) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/dao/selection/QuerySelectionHelper.java:222]: hadn't seen it and thought i was missing 🤦‍♂️ 
thx!

**Comment by metas-ts** [comment]
continuing with https://github.com/metasfresh/metasfresh/pull/14701

**Pull request closed without merging**